### PR TITLE
chore: [one click binding] gsheets query adaptor

### LINKED ISSUE #23255: Gsheets query generator for one click binding
Add support for gsheets query generator for one clickbinding epic.

### PR DESCRIPTION
**Pull request by vsvamsi1**
## Description
Developed Gsheets query generator for one click binding epic.

#### PR fixes following issue(s)
Fixes #23255

#### Type of change
- New feature (non-breaking change which adds functionality)
- Chore (housekeeping or task changes that don't impact user perception)

#### How Has This Been Tested?
- [x] Manual
- [x] Jest
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/WidgetQueryGenerators/GSheets/index.ts [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+    widgetConfig: WidgetQueryGenerationConfig,
+    formConfig: WidgetQueryGenerationFormConfig,
+  ): Record<string, object | string> | undefined {
+    const { update } = widgetConfig;
+
+    if (update) {
+      return {
+        type: QUERY_TYPE.UPDATE,
+        name: "Update_query",
+        //tableHeaderIndex
```
Comment (sbalaji1192) [inline app/client/src/WidgetQueryGenerators/GSheets/index.ts:150]: Can you add details to this comment?

**Review by sbalaji1192 — APPROVED** [review]

**Comment by chandannkumar** [comment]
Observations @vsvamsi1 
1. Not able to create a simple gsheet query via one click binding
https://www.loom.com/share/e516d58e97c34b1d9476a823422c74c1

2. 'Table header Index' field is very compact and also very close to 'Connect Data' button


3. Sheet name is overflowing from property pane, we should either wrap or display the name of sheet instead of sheet link


4. "Connect data" button must be inactive until Sheet name is selected, as it is a mandatory field

**Comment by chandannkumar** [comment]
Tested this PR and LGTM
- Verified all callouts
- Tried and tested with different Gsheets
- Overall flow of one-click binding

File: app/client/src/components/editorComponents/WidgetQueryGeneratorForm/DatasourceSpecificControls/GoogleSheetControls/SheetsDropdown/useSheets.tsx [hunk 2]
```diff
@@ -76,6 +76,7 @@ export function useSheets() {
     error: sheets?.error,
     options,
     isLoading,
+    labelText: "Select sheet from " + config.table,
```
Comment (sbalaji1192) [inline app/client/src/components/editorComponents/WidgetQueryGeneratorForm/DatasourceSpecificControls/GoogleSheetControls/SheetsDropdown/useSheets.tsx:79]: There is already a label property

File: app/client/src/components/editorComponents/WidgetQueryGeneratorForm/CommonControls/TableOrSpreadsheetDropdown/useTableOrSpreadsheet.tsx [hunk 3]
```diff
@@ -152,6 +152,7 @@ export function useTableOrSpreadsheet() {
     error: isGoogleSheetPluginDS(selectedDatasourcePluginPackageName)
       ? spreadSheets?.error
       : datasourceStructure?.error?.message,
+    labelText: `Select ${fieldName} from ${selectedDatasource?.name}`,
     label: (
```
[the remark this answers is not in the document]
Comment (vsvamsi1) [reply app/client/src/components/editorComponents/WidgetQueryGeneratorForm/CommonControls/TableOrSpreadsheetDropdown/useTableOrSpreadsheet.tsx:156]: In labelText is pure text that would go into the tooltip, label had Bold jsx components which i did not want in the tooltip. So for that reason I had another I added another property.

File: app/client/src/components/editorComponents/WidgetQueryGeneratorForm/DatasourceSpecificControls/GoogleSheetControls/SheetsDropdown/useSheets.tsx (hunk 2 above)
Comment (vsvamsi1) [reply app/client/src/components/editorComponents/WidgetQueryGeneratorForm/DatasourceSpecificControls/GoogleSheetControls/SheetsDropdown/useSheets.tsx:79]: In labelText its a pure text that would go into the tooltip, label had Bold jsx components which i did not want in the tooltip. So for that reason I had another I added another property.

**Review by sbalaji1192 — APPROVED** [review]

**Pull request merged**